kratos/utilities/sensitivity_builder.cpp

```diff
@@ -684,71 +599,89 @@ void SensitivityBuilder::CalculateNonHistoricalSensitivities(
     const ProcessInfo& rProcessInfo,
     double ScalingFactor)
 {
-    using sensitivity_builder_cpp::CalculateNonHistoricalSensitivities;
-    CalculateNonHistoricalSensitivities(
-        rVariables, rConditions, rResponseFunction, rProcessInfo, ScalingFactor);
+    using namespace sensitivity_builder_cpp;
+
+    const auto& r_variables_list = GetVariableLists(rVariables);
+    SensitivityBuilderScheme scheme;
+
+    CalculateNonHistoricalSensitivities(r_variables_list, rConditions, rResponseFunction,
+                                        scheme, rProcessInfo, ScalingFactor);
+}
+
+void SensitivityBuilder::CalculateNonHistoricalSensitivities(
+    const TSensitivityVariables& rVariables,
+    ModelPart::ConditionsContainerType& rConditions,
+    AdjointResponseFunction& rResponseFunction,
+    SensitivityBuilderScheme& rSensitivityBuilderScheme,
+    const ProcessInfo& rProcessInfo,
+    double ScalingFactor)
+{
+    using namespace sensitivity_builder_cpp;
+
+    ExecuteFunctor<CalculateNonHistoricalSensitivitiesFunctor>(
+        rVariables, rConditions, rResponseFunction, rSensitivityBuilderScheme,
+        ParallelEnvironment::GetDefaultDataCommunicator(), rProcessInfo, ScalingFactor);
```

Comment:
hm, why cannot you get it from the ModelPart?

Reply:
The existing interface does not pass the `ModelPart` so I had to use default communicator to keep the same interface.

Reply:
but the SensitivityBuilder seems to save the ModelPart as a member ?

Reply:
Unfortunately, those methods are implemented as static :/

Reply:
then I would drop the `static`, there is anyway no advantage and it is now what we usually do

Reply:
What if I use a deprecation msg for the static method, and introduce member method? I am afraid I will break someone else's work in some branch if I remove this static method :/

Reply:
hm I see

Seems to me like the interface is a bit messy :/

Well then lets leave this one for the next update